LabExT/Measurements/InsertionLossSteppedSweep.py

```diff
@@ -0,0 +1,174 @@
+            'averaging time': MeasParamFloat(value=0.1, unit='s'),
+            # laser power in dBm
+            'laser power': MeasParamFloat(value=6.0, unit='dBm'),
+            # range of the power meter in dBm
+            'powermeter range': MeasParamFloat(value=10.0, unit='dBm'),
+            # let the user give some own comment
+            'users comment': MeasParamString(value=''),
+        }
+
+    @staticmethod
+    def get_wanted_instrument():
+        return ['Laser', 'Power Meter']
+
+    def algorithm(self, device, data, instruments, parameters):
+        # get the parameters
+        start_lambda = parameters.get('wavelength start').value
+        end_lambda = parameters.get('wavelength stop').value
+        center_wavelength = (start_lambda + end_lambda) / 2
+        lambda_step = parameters.get('wavelength step').value
+        stabilization_time = parameters.get('stabilization time').value # EM
+        averaging_time = parameters.get('averaging time').value # EM
+        laser_power = parameters.get('laser power').value
+        pm_range = parameters.get('powermeter range').value
+
+        # get instrument pointers
+        self.instr_pm = instruments['Power Meter']
+        self.instr_laser = instruments['Laser']
+
+        # open connection to Laser & PM
+        self.instr_laser.open()
+        self.instr_pm.open()
+
+        # clear errors
+        self.instr_laser.clear()
+        self.instr_pm.clear()
+
+        # Ask minimal possible wavelength
+        min_lambda = float(self.instr_laser.min_lambda)
+
+        # Ask maximal possible wavelength
+        max_lambda = float(self.instr_laser.max_lambda)
+
+        # change the minimal & maximal wavelengths if necessary
+        if start_lambda < min_lambda or start_lambda > max_lambda:
+            start_lambda = min_lambda
+            parameters['wavelength start'].value = start_lambda
+            self.logger.warning('start_lambda has been changed to smallest possible value ' + str(min_lambda))
+
+        if end_lambda > max_lambda or end_lambda < min_lambda:
+            end_lambda = max_lambda
+            parameters['wavelength stop'].value = end_lambda
+            self.logger.warning('end_lambda has been changed to greatest possible value ' + str(max_lambda))
+
+        # write the measurement parameters into the measurement settings
+        for pname, pparam in parameters.items():
+            data['measurement settings'][pname] = pparam.as_dict()
+
+        # Laser settings
+        self.instr_laser.unit = 'dBm'
+        self.instr_laser.power = laser_power
+        self.instr_laser.wavelength = start_lambda
+        number_of_points = int((end_lambda - start_lambda) / lambda_step) + 1
+
+        # PM settings
+        self.instr_pm.averagetime = averaging_time
+        self.instr_pm.wavelength = center_wavelength
+        self.instr_pm.range = pm_range
+        self.instr_pm.unit = 'dBm'
+
+        time.sleep(1) # allow laser to move to start wavelength
```

Comment:
Please double check if this timed wait is really necessary. Timed waits should not be used, instead poll the instrument if the desired wavelength is reached. Most often, this can be done with something like `OPC?` but I suspect Keysight offers a specific command for checking if wavelength is stable?